gdk/commands/component/PublishCommand.py

```diff
@@ -136,27 +136,43 @@ def _publish_component_version(self, component_name, component_version):
         PublishRecipeTransformer(self.project_config).transform()
 
         logging.info(f"Creating a new greengrass component {component_name}-{component_version}")
-        self.greengrass_client.create_gg_component()
+        self.greengrass_client.create_gg_component(self.project_config["publish_recipe_file"])
 
     def upload_artifacts_s3(self) -> None:
         """
         Uploads all the artifacts from component artifacts build folder to s3 bucket.
 
         Raises an exception when the request is not successful.
         """
-        bucket = self.project_config["bucket"]
-        region = self.project_config["region"]
+        _bucket = self.project_config["bucket"]
+
+        build_component_artifacts = list(self.project_config["gg_build_component_artifacts_dir"].iterdir())
+
+        if not build_component_artifacts:
+            logging.info("No artifacts found in the component build folder. Skipping the artifact upload step")
+            return
+
         logging.info(
-            f"Uploading component artifacts to S3 bucket: {bucket}. If this is your first time using this bucket, add the"
-            " 's3:GetObject' permission to each core device's token exchange role to allow it to download the component"
-            f" artifacts. For more information, see {utils.doc_link_device_role}."
+            (
+                "Uploading component artifacts to S3 bucket: %s. If this is your first time using this bucket, add the"
+                " 's3:GetObject' permission to each core device's token exchange role to allow it to download the component"
```

Comment:
Is there any way to automate this?

Reply:
No, we don't know what role they're setup to use, and we don't want to blindly add wide permissions either.

Reply:
Cool, it's a hard thing to remember and find. I'm glad we have a comment here at least. :)